### PR TITLE
Add `min` and `max` to reserved words list

### LINKED ISSUE #234: `max` and `min` as settings names behave unexpectedly
This is on version 2.0.0

Using `max` and `min` appear to be reserved words and will behave unexpectedly.
```yaml
settings_group:
  max: "max"
  min: "min"
  other: "other"
```

## Expected
```ruby
> Settings.settings_group.max
=> "max"
> Settings.settings_group.min
=> "min"
> Settings.settings_group.other
=> "other"
```

## Actual
```ruby
> Settings.settings_group.max
=> [:other, "other"]
> Settings.settings_group.min
=> [:max, "max"]
> Settings.settings_group.other
=> "other"
```

### PR DESCRIPTION
**Pull request by supremebeing7**
Fixes #234

### CODE REVIEW

**Comment by supremebeing7** [comment]
I'll take a look at the tests when I have a minute and see about updating those as well.

**Comment by pkuczynski** [comment]
Some tests would be awesome!

File: README.md [hunk 1]
```diff
@@ -453,6 +453,12 @@ gem install bundler -v 1.17.3
 gem install appraisal
 ```
 
+Bundle
+
+```bash
+bundle install
+```
```
Comment (supremebeing7) [inline README.md]: `appraisal install` didn't work for me until I did this.

**Comment by supremebeing7** [comment]
OK, tests are up, all green locally.

**Comment by supremebeing7** [comment]
Looks like the travis failure is present on master

**Comment by pkuczynski** [comment]
@supremebeing7 thanks for working on this! Maybe you would like to become a contributor?

**Comment by supremebeing7** [comment]
@pkuczynski Happy to help when I have time! But that doesn't happen very often.

**Comment by pkuczynski** [comment]
Great! Every little help counts. Just added you to the repo

**Pull request merged**